src/v/security/audit/audit_log_manager.cc

```diff
@@ -176,19 +190,23 @@ ss::future<> audit_client::initialize() {
     }
 }
 
-ss::future<> audit_client::configure() {
-    try {
-        auto config = co_await kafka::client::create_client_credentials(
-          *_controller,
-          config::shard_local_cfg(),
-          _client.config(),
-          audit_principal);
-        set_client_credentials(*config, _client);
+ss::future<> audit_client::set_client_credentials() {
+    /// Set ephemeral credential
+    auto& frontend = _controller->get_ephemeral_credential_frontend().local();
+    auto pw = co_await frontend.get(audit_principal);
+    if (pw.err != cluster::errc::success) {
+        throw std::runtime_error(fmt::format(
+          "Failed to fetch credential for principal: {}", audit_principal));
+    }
```

Comment:
Should this be wrapped in some sort of retry logic?  Is it ever possible that the ephemeral credential front end isn't up and running yet (for whatever reason I can't think of)?  Is this how pp/sr do this?

Reply:
It is, check out `audit_client::initialize()`

Reply:
thanks!